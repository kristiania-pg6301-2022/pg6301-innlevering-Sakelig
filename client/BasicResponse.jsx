import { Link } from "react-router-dom";
import * as React from "react";

export function BasicResponse({ response }) {
  return (
    <div>
      {response ? <h1>Det var rikitg!</h1> : <h1>Det var feil!</h1>}
      <Link to={"/question"}> nytt spørmål</Link>
      <br />
      <Link to={"/"}> back home</Link>
    </div>
  );
}
