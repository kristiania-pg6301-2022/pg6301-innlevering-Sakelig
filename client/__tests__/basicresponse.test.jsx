import ReactDOM from "react-dom";
import React from "react";
import { BasicResponse } from "../BasicResponse";
import { MemoryRouter } from "react-router-dom";

describe("BasicresponsePage", () => {
  it("should show true answer page", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <BasicResponse response={true} />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("should show wrong answer page", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <BasicResponse response={false} />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });
});
