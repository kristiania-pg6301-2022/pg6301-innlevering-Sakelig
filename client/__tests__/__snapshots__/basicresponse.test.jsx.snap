// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicresponsePage should show true answer page 1`] = `"<div><h1>Det var rikitg!</h1><a href=\\"/question\\"> nytt spørmål</a><br><a href=\\"/\\"> back home</a></div>"`;

exports[`BasicresponsePage should show wrong answer page 1`] = `"<div><h1>Det var feil!</h1><a href=\\"/question\\"> nytt spørmål</a><br><a href=\\"/\\"> back home</a></div>"`;
